### PR TITLE
bumping grpc packages

### PR DESCRIPTION
**Pull request by brettsam**
### Issue describing the changes in this PR

Moving to latest grpc versions.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: extensions/Worker.Extensions.Rpc/src/GrpcHttpClientBuilderExtensions.cs [hunk 1]
```diff
@@ -22,11 +22,6 @@ public static class GrpcHttpClientBuilderExtensions
         /// Configures a <see cref="IHttpClientBuilder" /> to communicate with the functions host.
         /// </summary>
         /// <param name="builder">The original builder for call chaining.</param>
-        /// <remarks>
-        /// When using Grpc.Net.ClientFactory 2.54.0-pre1 or earlier, use a <see cref="GrpcClientFactoryOptions"/>
-        /// configure accepting overload of <c>AddGrpcClient</c>, such as
-        /// <see cref="GrpcClientServiceExtensions.AddGrpcClient{TClient}(IServiceCollection, Action{GrpcClientFactoryOptions})"/>.
-        /// </remarks>
```
Comment (brettsam) [inline extensions/Worker.Extensions.Rpc/src/GrpcHttpClientBuilderExtensions.cs:29]: @jviau -- is this the correct update? (basing on the test comment below)

File: test/Worker.Extensions.Rpc.Tests/GrpcHttpClientBuilderExtensionsTests.cs [hunk 2]
```diff
@@ -21,22 +21,7 @@ public void ConfigureForFunctionsHostGrpc_Configure_SetsUri()
...
-            {
-                ConfigureForFunctionsHostGrpc(s => s.AddGrpcClient<CallInvokerExtractor>());
-            }
-            catch (InvalidOperationException ex)
-            {
-                exception = ex;
-            }
-
-            Assert.NotNull(exception);
-        }
```
Comment (brettsam) [inline test/Worker.Extensions.Rpc.Tests/GrpcHttpClientBuilderExtensionsTests.cs:39]: @jviau -- this appears to be fixed now; i *think* this is the correct fix to the test?

File: src/DotNetWorker.Grpc/DotNetWorker.Grpc.csproj [hunk 3]
```diff
@@ -22,8 +22,9 @@
 
   <ItemGroup>
     <PackageReference Include="Azure.Core" Version="1.10.0" />
-    <PackageReference Include="Google.Protobuf" Version="3.21.7" />
-    <PackageReference Include="Grpc.Tools" Version="2.49.1">
+    <PackageReference Include="Google.Protobuf" Version="3.23.3" />
+    <PackageReference Include="Grpc.Core" Version="2.46.6" />
```
Comment (kshyju) [inline src/DotNetWorker.Grpc/DotNetWorker.Grpc.csproj:26]: We have a new entry to add `Grpc.Core` with this version bump change?

**Review by kshyju — APPROVED** [review]

**Comment by kshyju** [comment]
#1714 can be closed?

File: src/DotNetWorker/DotNetWorker.csproj [hunk 4]
```diff
@@ -8,7 +8,7 @@
     <AssemblyName>Microsoft.Azure.Functions.Worker</AssemblyName>
     <RootNamespace>Microsoft.Azure.Functions.Worker</RootNamespace>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
-    <MinorProductVersion>16</MinorProductVersion>
+    <MinorProductVersion>17</MinorProductVersion>
```
Comment (liliankasem) [inline src/DotNetWorker/DotNetWorker.csproj:11]: Is it okay if we include the deferred binding changes in the .17 release or does this need to go out asap?

File: src/DotNetWorker.Grpc/DotNetWorker.Grpc.csproj (hunk 3 above)
Comment (brettsam) [reply src/DotNetWorker.Grpc/DotNetWorker.Grpc.csproj:26]: Hmm I don't know what happened there. Let me back that part out.

File: src/DotNetWorker/DotNetWorker.csproj (hunk 4 above)
Comment (liliankasem) [reply src/DotNetWorker/DotNetWorker.csproj:11]: I take this back, we can do deferred binding in 18

**Review by liliankasem — APPROVED** [review]

**Review by jviau — APPROVED** [review]

**Pull request merged**